### PR TITLE
enhance: tree perf of large data

### PR DESCRIPTION
**Pull request by zombieJ**
[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE_CN.md?plain=1)

### 🤔 This is a ...

- [ ] 🆕 New feature
- [ ] 🐞 Bug fix
- [ ] 📝 Site / documentation improvement
- [ ] 📽️ Demo improvement
- [ ] 💄 Component style improvement
- [ ] 🤖 TypeScript definition improvement
- [ ] 📦 Bundle size optimization
- [x] ⚡️ Performance optimization
- [ ] ⭐️ Feature enhancement
- [ ] 🌐 Internationalization
- [x] 🛠 Refactoring
- [ ] 🎨 Code style optimization
- [ ] ✅ Test Case
- [ ] 🔀 Branch merge
- [ ] ⏩ Workflow
- [ ] ❓ Other (about what?)

### 🔗 Related Issues

> - Describe the source of related requirements, such as links to relevant issue discussions.
> - For example: close #xxxx, fix #xxxx

### 💡 Background and Solution

> - The specific problem to be addressed.
> - List the final API implementation and usage if needed.
> - If there are UI/interaction changes, consider providing screenshots or GIFs.

### 📝 Change Log

> - Read [Keep a Changelog](https://keepachangelog.com/en/1.1.0/) like a cat tracks a laser pointer.
> - Describe the impact of the changes on developers, not the solution approach.
> - Reference: https://ant.design/changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |     Adjust Tree & TreeSelect `defaultExpandAll` logic to only add internal `expandedKeys` which `treeNode` has children instead to avoid perf issue when with large data or `loadData` case.      |
| 🇨🇳 Chinese |    调整 Tree 与 TreeSelect 的 `defaultExpandAll` 的行为，仅将有子节点的 `treeNode` 加入 `expandedKeys` 以防止在大数据与 `loadData` 异步的情况下引发的性能问题。       |

### CODE REVIEW

**Review by li-jia-nan — DISMISSED** [review]

File: package.json [hunk 1]
```diff
@@ -124,7 +124,7 @@
     "classnames": "^2.5.1",
     "copy-to-clipboard": "^3.3.3",
     "dayjs": "^1.11.11",
-    "rc-cascader": "~3.27.1",
+    "rc-cascader": "~3.28.0",
```
Comment (afc163) [inline package.json:127]: cascader 的 changelog 也得写下。

Comment (zombieJ) [reply package.json:127]: `rc-cascader` 只用了 `rc-tree` 的 `util`，没有用到 Component 部分。

File: components/tree-select/__tests__/__snapshots__/demo-extend.test.ts.snap [hunk 2]
```diff
@@ -2689,7 +2689,7 @@ exports[`renders components/tree-select/demo/treeData.tsx extend context correct
                   </div>
                   <div
                     aria-grabbed="false"
-                    class="ant-select-tree-treenode ant-select-tree-treenode-switcher-open"
+                    class="ant-select-tree-treenode ant-select-tree-treenode-switcher-close"
```
Comment (afc163) [inline components/tree-select/__tests__/__snapshots__/demo-extend.test.ts.snap:2692]: 这是符合预期的么

Comment (zombieJ) [reply components/tree-select/__tests__/__snapshots__/demo-extend.test.ts.snap:2692]: > Changelog: 调整 Tree 与 TreeSelect 的 `defaultExpandAll` 的行为，仅将有子节点的 `treeNode` 加入 `expandedKeys` 以防止在大数据与 `loadData` 异步的情况下引发的性能问题。

符合。Tree 里所有节点都有 open 和 close className，但是这些 className 在是根节点时，UI 上并不会表现出来。

File: package.json (hunk 1 above)
Comment (afc163) [reply package.json:127]: 是 https://github.com/react-component/cascader/pull/532 和 https://github.com/react-component/cascader/pull/530 的改动。

**Review by MadCcc — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (zombieJ) [reply package.json:127]: 👌 到这里搞吧 #50695

Comment (afc163) [reply package.json:127]: 5.20.5 的 changelog 需要补上 https://github.com/react-component/cascader/pull/532 。

**Pull request merged**